pkg/controller/elasticsearch/driver/upgrade_predicates.go

```diff
@@ -222,6 +222,14 @@ var predicates = [...]Predicate{
 			if err != nil {
 				return false, err
 			}
+			_, healthyNode := context.healthyPods[candidate.Name]
+			// Get the number of expected masters
+			expectedMasters := len(context.masterNodesNames)
+			healthyPods := len(context.healthyPods)
+			if healthyNode && healthyPods == 1 && expectedMasters == 1 && len(deletedPods) == 0 {
```

Comment:
I am not sure that this condition is sufficient to detect a one node cluster, as you could have a single master cluster with a bunch of unhealthy Pods that hold data nodes.